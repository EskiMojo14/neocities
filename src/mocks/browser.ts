import { http, HttpResponse } from "msw";
import { setupWorker } from "msw/browser";
import pages from "./graph.json";

export const worker = setupWorker(
  http.get(/localhost:\d+\/___graph.json/, () => HttpResponse.json(pages)),
);
