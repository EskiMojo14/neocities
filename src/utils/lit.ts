import type { Properties } from "csstype";
import type { CSSResultGroup } from "lit";
import { css } from "lit";
import type { StyleInfo } from "lit-html/directives/style-map.js";
import { styleMap as _styleMap } from "lit-html/directives/style-map.js";

declare module "csstype" {
  // eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
  interface Properties {
    [variable: `--${string}`]: StyleInfo[string];
  }
}

export function styleMap(props: Properties) {
  return _styleMap(props as never);
}

export function rtl(styles: CSSResultGroup): CSSResultGroup {
  return css`
    :dir(rtl) &,
    &:dir(rtl) {
      ${styles};
    }
  `;
}

export function ensureDir(dir: string): "ltr" | "rtl" | "auto" {
  return dir === "rtl" || dir === "ltr" ? dir : "auto";
}
