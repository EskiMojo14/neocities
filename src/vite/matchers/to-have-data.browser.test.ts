import { html } from "lit";
import { expect, it } from "vitest";
import { page } from "vitest/browser";

it("should pass without value", () => {
  page.render(html`<div data-foo="bar">text</div>`);
  expect(page.getByText("text").element()).toHaveData("foo");
});

it("should fail without value", () => {
  page.render(html`<div data-foo="bar">text</div>`);
  expect(() => {
    expect(page.getByText("text").element()).toHaveData("baz");
  }).toThrowErrorMatchingSnapshot();
  expect(() => {
    expect(page.getByText("text").element()).not.toHaveData("foo");
  }).toThrowErrorMatchingSnapshot();
});

it("should pass with value", () => {
  page.render(html`<div data-foo="bar">text</div>`);
  expect(page.getByText("text").element()).toHaveData("foo", "bar");
});

it("should fail with value", () => {
  page.render(html`<div data-foo="bar">text</div>`);
  expect(() => {
    expect(page.getByText("text").element()).toHaveData("foo", "baz");
  }).toThrowErrorMatchingSnapshot();
  expect(() => {
    expect(page.getByText("text").element()).not.toHaveData("foo", "bar");
  }).toThrowErrorMatchingSnapshot();
});

it("should fail with invalid element", () => {
  expect(() => {
    expect(null).toHaveData("foo");
  }).toThrowErrorMatchingSnapshot();
});
