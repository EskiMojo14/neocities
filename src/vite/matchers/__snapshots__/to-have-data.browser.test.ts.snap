// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should fail with invalid element 1`] = `[Error: Expected element to be an HTMLElement]`;

exports[`should fail with value 1`] = `
[Error: [2mexpect([22m[31melement[39m[2m).[22mtoHaveData[2m([22m[32mkey[39m[2m, [22m[32mvalue[39m[2m)[22m
Expected element with data attribute "foo": [32m"baz"[39m
Received element with data attribute "foo": [31m"bar"[39m]
`;

exports[`should fail with value 2`] = `
[Error: [2mexpect([22m[31melement[39m[2m).[22mnot[2m.[22mtoHaveData[2m([22m[32mkey[39m[2m, [22m[32mvalue[39m[2m)[22m
Expected element without data attribute "foo": [32m"bar"[39m
Received element with data attribute "foo": [31m"bar"[39m]
`;

exports[`should fail without value 1`] = `
[Error: [2mexpect([22m[31melement[39m[2m).[22mtoHaveData[2m([22m[32mkey[39m[2m)[22m
Expected element with data attribute "baz"
Received element without data attribute "baz"]
`;

exports[`should fail without value 2`] = `
[Error: [2mexpect([22m[31melement[39m[2m).[22mnot[2m.[22mtoHaveData[2m([22m[32mkey[39m[2m)[22m
Expected element without data attribute "foo"
Received element with data attribute "foo"]
`;
